src/main/java/rolling/Roller.java

```diff
@@ -123,60 +117,57 @@ private static Roll createRoll(Dice dice) {
      * @param opportunities    If opportunities are enabled
      * @param doomHandler      Function to handle increases in doom from opportunities
      * @param plotPointHandler Function to handle changes in plot points from plot dice and opportunities
-     * @return A try that attempts to contain a pair with the rolls and the modifiers but will return an exception there
-     * is an error modifying plot points or doom points
+     * @return An object with the result and the changes in plot and doom points
      */
-    public static CompletableFuture<Triple<Result, Integer, Integer>> handleOpportunities(Result result, int discount, boolean opportunities, IntFunction<Integer> doomHandler, IntFunction<CompletableFuture<Integer>> plotPointHandler) {
+    public static PointChange handlePoints(Result result, int discount, boolean opportunities, IntFunction<Integer> doomHandler, IntFunction<CompletableFuture<Integer>> plotPointHandler) {
         int doomCount = opportunities ? result.getOpportunities() : 0;
         int plotPointsSpent = result.getPlotDiceCost() - discount;
         if (doomCount > 0) {
             plotPointsSpent--;
         }
         final int newDoomValue = doomHandler.apply(doomCount);
         final CompletableFuture<Integer> newPlotPointFuture = plotPointHandler.apply(plotPointsSpent * -1);
-        return newPlotPointFuture.thenApply(newPlotPointValue -> Triple.of(result, newDoomValue, newPlotPointValue));
+        return newPlotPointFuture.thenApply(newPlotPointValue -> new PointChange(result, newDoomValue, newPlotPointValue)).join();
```

Comment:
Why switching to joins?

Reply:
The process can't continue until this step is done, and join should be used since it throws an unchecked exception which I believe try covers, unlike .get()